spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/ta4j/BarContext.java

```diff
@@ -70,27 +80,73 @@
         this.startTime = newStartTime;
         this.endTime = startTime.plus(duration);
         this.close = newClose != null ? newClose.doubleValue() : 0;
-
-        this.low = newLow != null ? newLow.doubleValue() : close;
-        this.high = newHigh != null ? newHigh.doubleValue() : close;
         this.open = newOpen != null ? newOpen.doubleValue() : close;
-        this.volume = newVolume != null ? newVolume.doubleValue() : 0;
+
+        this.low = Math.min(close, open);
+        this.high = Math.max(close, open);
+
+
+        calculateInitialVolume(newVolume);
+    }
+
+    private void calculateInitialVolume(final Number newVolume) {
+        if (newVolume == null) {
+            return;
+        }
+        if (isDurationMoreThanDay()) {
+            this.volume = newVolume.doubleValue();
+        } else {
+            this.initialDayVolume = newVolume.doubleValue();
+        }
     }
 
     public boolean isAfter(final ZonedDateTime timestamp) {
         return timestamp.isAfter(endTime.minus(Duration.ofSeconds(1)));
     }
 
+    /**
+     * The contract of the update call is that it is called without timestamp
+     * and is always within bounds of one bar duration.
+     *
+     * @param newClose  new close - mandatory
+     * @param newVolume new volume to be updated (might be null)
+     */
     @SuppressWarnings("checkstyle:AvoidInlineConditionals")
-    public void update(final Number newLow, final Number newHigh, final Number newClose, final Number newVolume) {
+    public void update(final Number newClose, final Number newVolume) {
         if (newClose == null) {
             throw new IllegalArgumentException("Cannot update bar context without at least specifying close price");
         }
         close = newClose.doubleValue();
-        low = Math.min(low, newLow == null ? close : newLow.doubleValue());
-        high = Math.max(high, newHigh == null ? close : newHigh.doubleValue());
+        low = Math.min(low, close);
+        high = Math.max(high, close);
+
+        calculateVolume(newVolume);
+        updatesReceived++;
+    }
+
+    private void calculateVolume(final Number newVolume) {
+        if (newVolume == null) {
+            return;
+        }
+        if (!isDurationMoreThanDay() && initialDayVolume >= 0) {
+            volume = newVolume.doubleValue() - initialDayVolume;
+        } else {
+            volume = volume + newVolume.doubleValue();
+        }
+    }
+
+    /**
+     * Gets the {@link Bar} object from the context.
+     * @return the {@link Bar}
+     */
+    @SuppressWarnings("checkstyle:AvoidInlineConditionals")
+    public Bar toBar() {
+        return new BaseBar(duration, endTime, open, high, low, close,
+                volume == 0 && updatesReceived == 0 ? initialDayVolume : volume);
```

Comment:
The initial day volume is the volume that was available when STARTING the bar, so  it should not be used to create the current bar? The volume of our bar should be 0 if we didnt receive updates I would say